tests/returntypes/transforms/test_batch_input_new.py

```diff
@@ -0,0 +1,278 @@
+    @qml.qnode(dev, diff_method="parameter-shift")
+    def circuit(input1, input2, weights):
+        Embedding(input1, wires=range(2), rotation="Y")
+        qml.RY(weights[0], wires=0)
+        qml.RY(input2[0], wires=0)
+        qml.RY(weights[1], wires=1)
+        return qml.expval(qml.PauliZ(1))
+
+    np.random.seed(42)
+    batch_size = 5
+    input1 = np.random.uniform(0, np.pi, (batch_size, 2), requires_grad=False)
+    input2 = np.random.uniform(0, np.pi, (4, 1), requires_grad=False)
+    weights = np.random.uniform(-np.pi, np.pi, (2,))
+
+    with pytest.raises(ValueError, match="Batch dimension for all gate arguments"):
+        res = circuit(input1, input2, weights)
+
+
+def test_batch_input_with_trainable_parameters_raises_error():
+    """Test that using the batch_input method with trainable parameters raises a ValueError."""
+    dev = qml.device("default.qubit", wires=2)
+
+    @qml.batch_input(argnum=0)
+    @qml.qnode(dev)
+    def circuit(input):
+        qml.RY(input, wires=1)
+        qml.CNOT(wires=[0, 1])
+        qml.RX(0.1, wires=0)
+        return qml.expval(qml.PauliZ(0) @ qml.PauliX(1))
+
+    batch_size = 3
+
+    input = np.linspace(0.1, 0.5, batch_size, requires_grad=True)
+
+    with pytest.raises(
+        ValueError,
+        match="Batched inputs must be non-trainable."
+        + " Please make sure that the parameters indexed by "
+        + "'argnum' are not marked as trainable.",
+    ):
+        circuit(input)
+
+
+def test_mottonenstate_preparation(mocker):
+    """Test that batching works for MottonenStatePreparation"""
+    dev = qml.device("default.qubit", wires=3)
+
+    @qml.batch_input(argnum=0)
+    @qml.qnode(dev)
+    def circuit(data, weights):
+        qml.templates.MottonenStatePreparation(data, wires=[0, 1, 2])
+        qml.templates.StronglyEntanglingLayers(weights, wires=[0, 1, 2])
+        return qml.probs(wires=[0, 1, 2])
+
+    batch_size = 3
+
+    # create a batched input statevector
+    data = np.random.random((batch_size, 2**3), requires_grad=False)
+    data /= np.linalg.norm(data, axis=1).reshape(-1, 1)  # normalize
+
+    # weights is not batched
+    weights = np.random.random((10, 3, 3), requires_grad=True)
+
+    spy = mocker.spy(circuit.device, "batch_execute")
+    res = circuit(data, weights)
+    assert res.shape == (batch_size, 2**3)
+    assert len(spy.call_args[0][0]) == batch_size
+
+    # check the results against individually executed circuits (no batching)
+    @qml.qnode(dev)
+    def circuit2(data, weights):
+        qml.templates.MottonenStatePreparation(data, wires=[0, 1, 2])
+        qml.templates.StronglyEntanglingLayers(weights, wires=[0, 1, 2])
+        return qml.probs(wires=[0, 1, 2])
+
+    indiv_res = []
+    for state in data:
+        indiv_res.append(circuit2(state, weights))
+    assert np.allclose(res, indiv_res)
+
+
+def test_multi_returns():
+    """Test that batching works for a simple circuit with multiple returns"""
```

Comment:
Added this test and the next two to test for multiple returns, shot vectors, and combined.

Reply:
Thanks for pointing this out!